tools/gitHooks/precommit.js

```diff
@@ -59,7 +82,18 @@ $ git stash pop
     await doLicenseCommit(changedFiles);
 
     // Generate API reports
-    await doApiReportsCommit();
+    await doApiReportsCommit(changedFiles);
+
+    // Diff staged changes against last commit. Don't do an empty commit.
+    const postDiff = await git.diff(['--cached']);
+    if (!postDiff) {
```

Comment:
It seems this condition will never be true since we are committing something from the staging area to begin with.

Reply:
There is a case I ran into while testing where if you only make formatting changes, and they are wrong, and Prettier reformats them back the way they were, there is no longer a diff between your changes and the last commit. I don't think this would ever happen on push but it could happen on a small commit if you did a lot of undos and don't realize the only remaining changes are formatting.